llvm/lib/Target/RISCV/RISCVAsmPrinter.cpp

```diff
@@ -704,19 +729,32 @@ void RISCVAsmPrinter::emitEndOfAsmFile(Module &M) {
 
       // Public symbols must be COMDATs so that they can be merged across
       // compilation units.  Private ones must not be.
-      auto *Section =
-          Entry.IsPublic
-              ? C.getELFSection(".compartment_imports", ELF::SHT_PROGBITS,
-                                ELF::SHF_ALLOC | ELF::SHF_GROUP, 0,
-                                Entry.ImportName, true)
-              : C.getELFSection(".compartment_imports", ELF::SHT_PROGBITS,
-                                ELF::SHF_ALLOC);
+      MCSectionELF *Section;
+
+      if (Entry.IsPublic) {
+        if (Entry.IsGlobal) {
+          Section = C.getELFSection(
+              ".compartment_imports." + Entry.Name, ELF::SHT_PROGBITS,
+              ELF::SHF_ALLOC | ELF::SHF_GROUP | ELF::SHF_WRITE, 0,
+              Entry.ImportName, true);
+        } else {
+          Section = C.getELFSection(
```

Comment:
This can be factored together with the call to `getELFSection` above it. It looks like they only differ in the `ELF_SHF_*` argument.

Reply:
adad1a7 refactors the two branches in a single statement.